### PR TITLE
Colormaps

### PR DESCRIPTION
**Pull request by doutriaux1**
@aashish24 @sankhesh I'm still teaking a few things, but somehow when merging master back in, meshfill animations test is now broken @sankhesh do you mind taking  a look at this while I fine tune this PR? Thanks.

### CODE REVIEW

**Comment by doutriaux1** [comment]
@aashish24 @chaosphere2112 @sankhesh this 20mn PR is finally ready for review. I still need help from @sankhesh for the meshfill animation which I think I broke when merging master in.
Basic idea of this PR: allow transparency on colors, allow to set the whole 256 colors on a vcs colormap, conversion tool from matplotlib cmaps to vcs ones and allow to set a color directly by its "name" or rgba value.
Goes with: https://github.com/UV-CDAT/uvcdat-testdata/pull/84

**Comment by sankhesh** [comment]
@doutriaux1 I'll look into it today. Are we allowed to commit changes today?

**Comment by chaosphere2112** [comment]
@sankhesh Yeah, email out from @aashish24 on Friday indicated freeze was 11:59PM tonight.

**Comment by jbeezley** [comment]
Yes, but which timezone? :stuck_out_tongue_winking_eye:

**Comment by chaosphere2112** [comment]
@jbeezley Let's shoot for UTC-11 :wink:

**Comment by aashish24** [comment]
> Yes, but which timezone?

In the email I referred to 11:59 PM PT

**Comment by aashish24** [comment]
Here is it again: 

As per our last meeting, the code freeze will happen on Monday night 11:59 PT. Please try to look at the bugs assigned to you for release 2.4. If for some reason, you think you cannot get to resolve them in time, please contact Dean, Charles, or myself.

**Comment by aashish24** [comment]
its common to have 20mn PR become 2hr PR :smiley:

**Comment by doutriaux1** [comment]
I know but this one went into a 2 weeks PR...

File: Packages/vcs/Lib/colormap.py [hunk 1]
```diff
@@ -231,12 +288,12 @@ def __init__(self, Cp_name, Cp_name_src='default'):
         # Note: See RGB_Table Class for "index" setting of the colormap entries
 
     # Set a colorcell RGB
-    def setcolorcell(self, index, red, green, blue):
+    def setcolorcell(self, index, red, green, blue, alpha=100.):
```
Comment (aashish24) [inline Packages/vcs/Lib/colormap.py]: why default to 100?

Comment (chaosphere2112) [inline Packages/vcs/Lib/colormap.py]: Legacy support. All of the old usages of this function would have generated opaque colors, no reason to change that behavior.

**Comment by aashish24** [comment]
@doutriaux1 at a high level, the changes makes sense to me. Only had one question but that's because of me not knowing certain things.

**Comment by aashish24** [comment]
LGTM :+1:

**Comment by aashish24** [comment]
@doutriaux1 is going to fix the merge conflict..

**Comment by doutriaux1** [comment]
I'm at loss why mac's still fail... Master built on my Mac when I reinstalled xcde command line

**Comment by doutriaux1** [comment]
Mac1 worked....  Arggggg going crazy....

**Comment by doutriaux1** [comment]
Mac1 worked....  Arggggg going crazy....

**Comment by doutriaux1** [comment]
Serious issues since merged master in. Will look at that asap

**Comment by dnadeau4** [comment]
I have trouble with my Mac as well.  My python(2.7.10) does not want to 
compile numpy.  Something about broken toolchain...

   RuntimeError: Broken toolchain: cannot link a simple C program. I
tried to delete everything run cmake and "make NUMPY" and it failed!!

**Comment by doutriaux1** [comment]
@dnadeau4 lab mac? or home mac? if lab mac please try to manually re-install xcode command line tools and try again it fixed it for me and it fixed it for macbot1...

**Comment by doutriaux1** [comment]
I spent my day undoing one at a time the various PR we merged over the last 2 weeks, to no success, I strongly suspect a lab patch issue, does anybody else has a non lab mac and could try to build on it? @potter2 ?

**Comment by dnadeau4** [comment]
I did not notice the lab udated my Xcode... Things seems better now that 
I reinstalled Xcode command lines.

Denis

**Comment by doutriaux1** [comment]
I don't think it's an XCode update, I think it was part of the `security update`....

**Comment by doutriaux1** [comment]
which means it's going to be patched back and forth for ever without us not really knowing what is being modified... I will try to contact them tomorrow...

**Comment by doutriaux1** [comment]
@sankhesh meshfill animations seem broken can you please help restoring this? thx.
@chaosphere2112 your magical code for pattern drawing is now broken :cry: sorry about this please help!

**Comment by doutriaux1** [comment]
right before the code freeze, I got 16mn to spare! I guess I will watch to see if my macbot are happy again with the build phase...

**Comment by doutriaux1** [comment]
All right the mac's seem to be fixed, nowci just need to update the baseline tomorrow.

**Comment by sankhesh** [comment]
@doutriaux1 Sure. I'll push changes to this pull request.

**Comment by doutriaux1** [comment]
thanks @sankhesh I'm in now, let me know if you need help and/or want to do a webex so we can work together on this.

**Comment by aashish24** [comment]
@doutriaux1 new changes looks reasonable to me. I am going to wait for the buildbot.

**Comment by aashish24** [comment]
@doutriaux1 @sankhesh Looks like the test is still failing (animated meshfill). What I am missing?

**Comment by aashish24** [comment]
@doutriaux1 @sankhesh I figured it out. Atleast one of the problem is missing baseline. I am adding it now

**Comment by aashish24** [comment]
@sankhesh @doutriaux1 @williams13  I fixed the animation test (really tiny fix). Not sure about the pattern failing test. @sankhesh can you look into it?

**Comment by doutriaux1** [comment]
@aashish24 that's why when I looked at the meshfill test bg was set to 1 (should have been `True` really but i will let it fly unless I need to trigger a new build). I need to push the mercator baseline which is why the test fails. Will fix tomorrow. In the mean time when you get in @aashish24 do you mind merging this branch into @sankhesh pending branches so that we can approve them all quickly when I get in tomorrow? Thanks

**Comment by doutriaux1** [comment]
mac seem to be broken again, I wonder if the lab scans re`patched` them again....

**Comment by aashish24** [comment]
@doutriaux1 test patterns is also failing. Earlier 4 tests were failing but now only three are failing on garant. Do you still want me to merge it?

File: Packages/vcs/Lib/VCS_validation_functions.py [hunk 2]
```diff
@@ -2019,3 +2036,31 @@ def _setvp(self, value):
...
+    return self._fillareacolors
+
+
+def _setfillareacolors(self, value):
+    if value is not None:
+        value = checkColorList(
+            self,
+            'fillareacolors',
+            value)
+    self._fillareacolors = value
```
Comment (sankhesh) [inline Packages/vcs/Lib/VCS_validation_functions.py]: We allow the user to set a null value?  Is that intentional?

**Comment by sankhesh** [comment]
@doutriaux1 :+1: LGTM

I'm looking at merging my two other branches.

**Comment by aashish24** [comment]
@sankhesh do you think the tests failing are not because of this branch?

File: Packages/vcs/Lib/VCS_validation_functions.py (hunk 2 above)
Comment (doutriaux1) [inline Packages/vcs/Lib/VCS_validation_functions.py]: Yes, I think for missing value color None used to mean transparent

Comment (sankhesh) [inline Packages/vcs/Lib/VCS_validation_functions.py]: Okay. But that shouldn't be the case anymore, right? Since we have an opacity field now.

**Comment by sankhesh** [comment]
@aashish24 

The tests are definitely failing because of this branch. But the failures require updating the baselines.

`vcs_test_click_info` is a weird one. Fails on some machines only.

**Comment by doutriaux1** [comment]
@sankhesh I think `vcs_test_click_info` and a few others are screen size dependent, @chaosphere2112 was going to look into this. BTW @chaosphere2112 you can now do a x.open(800,600) to get the correct window size.

**Comment by aashish24** [comment]
@doutriaux1 new baseline?

**Comment by doutriaux1** [comment]
@aashish24 I tihnk it's ready to merge.

**Comment by doutriaux1** [comment]
@aashish24 crunchy (RH6 FULL) died so it will never complete

**Comment by chaosphere2112** [comment]
@doutriaux1 @aashish24 Tests all pass for me and the bots look good; I'm merging.

**Pull request merged**